plasma_framework/contracts/src/utils/RLPReader.sol

```diff
@@ -135,102 +178,85 @@ library RLPReader {
+                let lengthLen := sub(byte0, 0xb7) // Number of bytes the actual length is
                 memPtr := add(memPtr, 1) // skip over the first byte
+                byte1 := byte(0, mload(memPtr)) // read the first byte to check for leading zeros
 
-                byte1 := byte(0, mload(memPtr))
-
-                /* 32 byte word size */
-                dataLen := div(mload(memPtr), exp(256, sub(32, byteLen))) // right shifting to get the len
-                decodedItemLengthUnsafe := add(dataLen, add(byteLen, 1))
+                // right shift to the correct position
+                dataLen := div(mload(memPtr), exp(256, sub(WORD_SIZE, lengthLen)))
+                decodedLength := add(dataLen, add(lengthLen, 1))
             }
-            // Check valid long string i.e. value of length > MAX_SHORT_LEN with no leading zeros
+            // Check that the length has no leading zeros
             require(byte1 != 0, "Invalid RLP encoding");
+            // Check that the value of length > MAX_SHORT_LEN
             require(dataLen > MAX_SHORT_LEN, "Invalid RLP encoding");
+            // Calculate the offset
+            offset = (byte0 - (STRING_LONG_START - 1)) + 1;
         } else if (byte0 < LIST_LONG_START) {
-            decodedItemLengthUnsafe = (byte0 - LIST_SHORT_START) + 1;
+            // Item is a short list (length <= 55 bytes)
+            decodedLength = (byte0 - LIST_SHORT_START) + 1;
+            offset = 1;
         } else {
+            // Item is a long list (length > 55 bytes)
             uint256 dataLen;
             uint256 byte1;
             // solhint-disable-next-line no-inline-assembly
             assembly {
-                let lengthLen := sub(byte0, 0xf7)
-                memPtr := add(memPtr, 1)
-
-                byte1 := byte(0, mload(memPtr))
+                let lengthLen := sub(byte0, 0xf7) // Number of bytes the actual length is
+                memPtr := add(memPtr, 1) // skip over the first byte
+                byte1 := byte(0, mload(memPtr)) // read the first byte to check for leading zeros
 
-                dataLen := div(mload(memPtr), exp(256, sub(32, lengthLen))) // right shifting to the correct length
-                decodedItemLengthUnsafe := add(dataLen, add(lengthLen, 1))
+                // right shift to the correct position
+                dataLen := div(mload(memPtr), exp(256, sub(WORD_SIZE, lengthLen)))
+                decodedLength := add(dataLen, add(lengthLen, 1))
             }
-            // Check valid long list i.e. value of length > MAX_SHORT_LEN with no leading zeros
+            // Check that the length has no leading zeros
             require(byte1 != 0, "Invalid RLP encoding");
+            // Check that the value of length > MAX_SHORT_LEN
             require(dataLen > MAX_SHORT_LEN, "Invalid RLP encoding");
+            // Calculate the offset
+            offset = (byte0 - (LIST_LONG_START - 1)) + 1;
```

Comment:
minor: same as previous comment it would easier to read if this is sth like how it is calculated previously in `add(lengthLen, 1)`.